docs/design/request-authentication.md

```diff
@@ -0,0 +1,384 @@
+# Request Authentication
+
+## Overview
+
+[Issue 336][] specifies the need for exposing a user-facing API to configure request authentication. Request
+authentication is defined as an authentication mechanism to be enforced by Envoy on a per-request basis. A connection
+will be rejected if it contains invalid authentication information, based on the `Authentication` API type proposed in
+this design document.
+
+Envoy Gateway leverages [Gateway API][] for configuring managed Envoy proxies. Gateway API defines core, extended, and
+implementation-specific API [support levels][] for implementors such as Envoy Gateway to expose features. Since
+implementing request authentication is not covered by `Core` or `Extended` APIs, an `Implementation-specific` API will
+be created for this purpose.
```

Comment:
The key point of this design is the Authentication API is meant to be an __implementation-specific filter__. The API does not follow the PolicyAttachment specification. The Authentication API defined by this design will allow App Devs to easily attach authentication to API endpoints. At a future date, I expect Envoy Gateway to support an AuthenticationPolicy API that follows the PolicyAttachment spec and allow Infra Admins to override or set defaults for fields defined by the Authentication API. The direction of this design will change if maintainers agree via #675 that PolicyAttachment is the preferred method for allowing App Devs to attach authentication to API endpoints.

cc: @youngnick @sunjayBhatia @skriss @AliceProxy @Xunzhuo

Reply:
Okay, I posted an update to #675 (https://github.com/envoyproxy/gateway/issues/675#issuecomment-1309767753), that I think should help understand this, but in short:
- I think Policy Attachment is _preferred overall_, **but**
- in this case, I think a custom filter is okay for now.

Reply:
Considering this resolved due to https://github.com/envoyproxy/gateway/pull/529#issuecomment-1299355946.